app/src/main/java/pm/gnosis/heimdall/data/repositories/impls/DefaultEnsRepository.kt

```diff
@@ -42,3 +45,12 @@ class DefaultEnsRepository @Inject constructor(
         private const val GET_RESOLVER = "0x0178b8bf"
     }
 }
+
+interface EnsNormalizer {
+    fun normalize(name: String): String
+}
+
+@Singleton
+class IDNEnsNormalizer @Inject constructor() : EnsNormalizer {
+    override fun normalize(name: String) = IDN.toASCII(name, IDN.USE_STD3_ASCII_RULES).toLowerCase(Locale.getDefault())
```

Comment:
the IDN.toASCII can throw IllegalArgumentException if the input is invalid domain name (for example, starts with two hyphens "--"). This will be handled, right?

Reply:
Errors are propagated